File: RestApi/HttpClient/Communication/HttpMatchesResponse.cs
using BasketballClient.Models;
using System.Collections.Generic;
using System.Net;

namespace BasketballClient.Communication
{
    public class HttpMatchesResponse
    {
        public IEnumerable<Match> Matches { get; set; }

        public bool IsSuccess { get; set; }

        public HttpStatusCode StatusCode { get; set; }

        public string ErrorMessage { get; set; }
    }
}

File: RestApi/JavaHttpClient/src/main/java/http/HttpMatchesResponse.java
package http;

import models.Match;

import java.util.List;

public class HttpMatchesResponse {
    private List<Match> matches;

    private Boolean isSuccess;

    private Integer statusCode;

    private String errorMessage;

    public HttpMatchesResponse() {}

    public List<Match> getMatches() {
        return matches;
    }

    public void setMatches(List<Match> matches) {
        this.matches = matches;
    }

    public Boolean getSuccess() {
        return isSuccess;
    }

    public void setSuccess(Boolean success) {
        isSuccess = success;
    }

    public Integer getStatusCode() {
        return statusCode;
    }

    public void setStatusCode(Integer statusCode) {
        this.statusCode = statusCode;
    }

    public String getErrorMessage() {
        return errorMessage;
    }

    public void setErrorMessage(String errorMessage) {
        this.errorMessage = errorMessage;
    }
}
